### PR TITLE
Use CCData to allocate GlobalFunctionCallData

### PR DESCRIPTION
**Pull request by r30shah**
Use CCData to allocate GlobalFunctionCallData

Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
@fjeremic @ymanton  Can I please get review on this change?

**Review by fjeremic — APPROVED** [review]
Looks good to me, thought delegating to Younes for the merge as he has more experience with the use of this API.

File: compiler/z/objectfmt/OMRELFJitCodeObjectFormat.cpp [hunk 1]
```diff
@@ -51,14 +54,16 @@ OMR::Z::ELFJitCodeObjectFormat::emitGlobalFunctionCall(TR::GlobalFunctionCallDat
...
    TR_ASSERT_FATAL(targetAddress != NULL, "Unable to make a call for n%dn %s as targetAddress for the Call is not found.", callNode->getGlobalIndex(), callNode->getOpCode().getName());
 
-   ccGlobalFunctionData *ccGlobalFunctionDataAddress =
-      reinterpret_cast<ccGlobalFunctionData *>(cg->allocateCodeMemory(sizeof(ccGlobalFunctionData), false));
-
-   if (!ccGlobalFunctionDataAddress)
+   OMR::CCData *codeCacheData = cg->getCodeCache()->manager()->getCodeCacheData();
+   OMR::CCData::index_t index;
+   
+   if (!(codeCacheData->put(NULL, sizeof(ccGlobalFunctionData), alignof(uintptr_t), NULL, index)))
```
Comment (ymanton) [inline compiler/z/objectfmt/OMRELFJitCodeObjectFormat.cpp:60]:
```suggestion
   if (!(codeCacheData->put(NULL, sizeof(ccGlobalFunctionData), alignof(ccGlobalFunctionData), NULL, index)))
```

**Review by ymanton — COMMENTED** [review]
LGTM. Just one suggestion, re: using the struct's natural alignment unless there's a compelling reason not to.

File: compiler/z/objectfmt/OMRELFJitCodeObjectFormat.cpp (hunk 1 above)
Comment (r30shah) [reply compiler/z/objectfmt/OMRELFJitCodeObjectFormat.cpp]: Thanks a lot for the suggestions, I think using the alignment of ccGlobalFunctionData works . Made changes in https://github.com/eclipse/openj9-omr/compare/40c06eea215a7f142827601c278ee37e63eb5e9a..bb0f01ada9bdfde593e7c36674e28c9ff28b7368

**Review by ymanton — APPROVED** [review]

**Pull request merged**